src/Administration/Resources/app/administration/src/module/sw-login/view/sw-login-login/sw-login-login.html.twig

```diff
@@ -24,60 +24,81 @@
+            {% block sw_login_login_password_field %}
+            <sw-password-field
+                v-model:value="password"
+                name="sw-field--password"
+                :label="$tc('sw-login.index.labelPassword')"
+                :placeholder="$tc('sw-login.index.placeholderPassword')"
+                :disabled="showLoginAlert"
+                required
+            />
+            {% endblock %}
 
-    <sw-checkbox-field
-        v-model:value="rememberMe"
-        :label="$tc('sw-login.index.labelKeepLoggedIn')"
-    />
+            <sw-checkbox-field
+                v-model:value="rememberMe"
+                :label="$tc('sw-login.index.labelKeepLoggedIn')"
+            />
 
-    <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
-    {% block sw_login_login_submit %}
-    <div class="sw-login__submit">
-        <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
-        {% block sw_login_login_forgot_password %}
-        <router-link
-            :to="{ name: 'sw.login.index.recovery' }"
-            class="sw-login__forgot-password-action"
-        >
-            {{ $tc('sw-login.index.forgottenPasswordLink') }}
-        </router-link>
-        {% endblock %}
+            <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
+            {% block sw_login_login_submit %}
+            <div class="sw-login__submit">
+                <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
+                {% block sw_login_login_forgot_password %}
+                <router-link
+                    :to="{ name: 'sw.login.index.recovery' }"
+                    class="sw-login__forgot-password-action"
+                >
+                    {{ $tc('sw-login.index.forgottenPasswordLink') }}
+                </router-link>
+                {% endblock %}
+
+                <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
+                {% block sw_login_login_submit_button %}
+                <sw-button
+                    :disabled="password.length <= 0 || username.length <= 0 || showLoginAlert"
+                    class="sw-login__login-action"
+                    variant="primary"
+                >
+                    {{ $tc('sw-login.index.buttonLogin') }}
+                </sw-button>
+                {% endblock %}
+            </div>
+            {% endblock %}
+        </template>
 
-        <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
-        {% block sw_login_login_submit_button %}
-        <sw-button
-            :disabled="password.length <= 0 || username.length <= 0 || showLoginAlert"
-            class="sw-login__login-action"
-            variant="primary"
-        >
-            {{ $tc('sw-login.index.buttonLogin') }}
-        </sw-button>
-        {% endblock %}
+        <template v-for="loginType in loginConfig.ssoProviders" :key="loginType.key">
```

Comment:
Looking only at this template, if `loginConfig.useDefault` was to be `false` but the `loginConfig.ssoProviders` for what ever reason had a length of 0, the customer would not see any possibility to login. Have you concidered this? Is this prevented in the backend somewhere?

Reply:
Yes, if no config is provided we always get ['useDefault' => true, 'ssoProviders' => []].